src/lsm/k_way_merge.zig

```diff
@@ -71,8 +74,8 @@ pub fn KWayMergeIterator(
                 it.keys[it.k] = stream_peek(context, stream_index) catch |err| switch (err) {
                     // On initialization, the streams should either have data already
                     // buffered up to peek or be empty and have no more values to produce.
-                    error.Drained => unreachable,
-                    error.Empty => continue,
+                    error.Again => unreachable,
+                    error.EOF => continue,
```

Comment:
Imo these new names are not an improvement.

- `EOF` stands for "End of File", which is not relevant to tigerbeetle.
- `Again` conveys less information than "drained". Unlike drained, it doesn't convey why no value was returned.

Reply:
I agree, `EOF` is a bad choice. But not sure about `Again` vs `Drained`.
The point of renaming was to reduce the confusion between `Empty` and `Drained`, since if it's drained, it's also empty, so the difference is not obvious. 

The name `Again` came up following Linux's `EAGAIN` idea: there's no data available, we need to do some external work and call it again. Before that, I was calling it `Refill`.

-----------

But now I realized that [we already had this conversation before](https://github.com/tigerbeetle/tigerbeetle/commit/bf23d70a188ac3c3233de6553c61cfbb32507012), where it was named `Buffering` 🙃, I guess we are just bikeshedding at this point, let's keep `Empty` and `Drained`.